core/drivers/gic.c

```diff
@@ -948,11 +950,34 @@ static void __maybe_unused gic_native_itr_handler(void)
 	iar = gic_read_iar(gd);
 	id = iar & GICC_IAR_IT_ID_MASK;
 
-	if (id <= gd->max_it)
+	if (id <= gd->max_it) {
 		interrupt_call_handlers(&gd->chip, id);
-	else
-		DMSG("ignoring interrupt %" PRIu32, id);
+		goto write_eoir;
+	} else if (IS_ENABLED(CFG_ARM_GICV3) && id >= 1020 && id <= 1023) {
+		/*
+		 * Special INTIDs
+		 * 1020: Interrupt expected to be handled at SEL1 or SEL2
+		 * 1021: Interrupt expected to be handled at NSEL1 or NSEL2
+		 * 1022: -(GICv3.3): Interrupt is an NMI
+		 *       -(Legacy): Group 1 interrupt to be signaled to the
+		 *        processing element and acknowledged using alias
+		 *        registers.
+		 * 1023: No pending interrupt with sufficient priority
+		 *       (spurious) or the highest priority pending interrupt is
+		 *       not appropriate for the current security state or
+		 *       interrupt group.
+		 */
+		DMSG("Special interrupt %"PRIu32, id);
+
+		return;
+	} else if (!IS_ENABLED(CFG_ARM_GICV3) &&
+		   id == GICV2_SPURIOUS_INTERRUPT_ID) {
```

Comment:
This is a "Special interrupt" too.

Reply:
Looking back at the documentation, even for GICV1/2, 1020-1023 are reserved IDs even if some are not used. Therefore, I'll stop discriminating GIC version and  try to group everything together.